### ISSUE TITLE
Fine-tuning Conformer

### ISSUE THREAD

**Issue by MelikaBahmanabadi** [NVIDIA/NeMo#3269]
Hi,
I want to fine-tune **stt_en_conformer_ctc_large** with Persian dataset. I used **conformer_ctc_char.yaml** from your example folder in repository. I changed the labels, too. Now when I want to run **trainer.fit(asr_model)** I have the following error:
`Interpolation key 'model.encoder.d_model' not found
    full_key: optim.sched.d_model
    object_type=dict`
Would you please help me how to resolve this error?
Thanks

_Environment overview_

> Environment location: Google Colab
> Method of NeMo install: !pip install nemo_toolkit[asr]
> NeMo version: r1.2.0
> Learning Rate: 1e-4

_Environment details_

> OS version : "Ubuntu20.04.3 LTS"
> PyTorch version : "1.10.0"

**Comment by MelikaBahmanabadi** [comment]
Hi,
Would you please share for me any note book for fine-tuning conformer?
Thanks

**Comment by VahidooX** [comment]
I am not sure if you can call the fit on a pretrained model again to train it on another dataset specially when the architectures of those models are different. The decoder of a char-based model is different from a BPE-based model. So in order to fine-tune it on another dataset, there are a bunch of ways. I think the easiest one is to use speech_to_text.py with the new config, add a bunch of lines to this script which loads the pretrained model and then load the weights of the encoder part into the new model, then call the fit method. Check here to get an idea how to load a pretrained model's weights into another one:

https://github.com/NVIDIA/NeMo/blob/main/nemo/core/classes/modelPT.py#L923

You may use someting like the following to just load the encoder part, not the whole model:

`new_asr_model.encoder.load_state_dict(restored_model.encoder.state_dict(), strict=False)`
